connect/client_test.go

```diff
@@ -6,13 +6,52 @@ import (
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	cluster_api "github.com/pomerium/zero-sdk/cluster"
 	"github.com/pomerium/zero-sdk/connect"
 	"github.com/pomerium/zero-sdk/token"
 )
 
+func TestConfig(t *testing.T) {
+	t.Parallel()
+
+	for _, tc := range []struct {
+		endpoint      string
+		connectionURI string
+		requireTLS    bool
+		expectError   bool
+	}{
+		{"", "", false, true},
+		{"http://", "", false, true},
+		{"https://", "", true, true},
+		{"localhost:8721", "", false, true},
+		{"http://localhost:8721", "dns:localhost:8721", false, false},
+		{"https://localhost:8721", "dns:localhost:8721", true, false},
+		{"http://localhost:8721/", "dns:localhost:8721", false, false},
+		{"https://localhost:8721/", "dns:localhost:8721", true, false},
+		{"http://localhost:8721/path", "dns:localhost:8721", false, true},
+		{"https://localhost:8721/path", "dns:localhost:8721", true, true},
```

Comment:
The `connectionURI` and `requireTLS` values here won't be used (because of the early return on line 45), right? Should we remove them?

Alternatively, would it be simpler to change the test assertions to something like this (assuming the last field were changed to the expected error message as a string):
```go
assert.Equal(t, tc.expectedError, err.Error(), "error string")
assert.Equal(t, tc.connectionURI, cfg.GetConnectionURI(), "connection uri")
assert.Equal(t, tc.requireTLS, cfg.RequireTLS(), "require tls")
```